File: cs/UCLA/libUCLA/UConfig.cs
namespace libUCLA {
    public class UConfig {
        public const string Proto = "tcp";

        /// <summary>
        /// Gets endpoint String
        /// </summary>
        public string Endpoint {
            get {
                return UConfig.Proto + "://" + source + ":" + port;
            }
        }

        private string source;
        private string port;

        /// <summary>
        /// UConfig constructor.
        /// </summary>
        /// <param name="source">IP address of host if client or interface if server(http://www.crossroads.io/1-0:xs-tcp).</param>
        /// <param name="port">Port number (as string).</param>
        public UConfig(string source, string port) {
            this.source = source;
            this.port = port;
        }
    }
}

File: java/libUCLA/src/libUCLA/UConfig.java
package libUCLA;

public class UConfig {

    public static final String Proto = "tcp";
    private String source;
    private String port;

    /**
     * @return Endpoint String
     */
    public String getEndpoint() {
        return UConfig.Proto + "://" + source + ":" + port;
    }

    /**
     * UConfig constructor.
     *
     * @param source IP address of host if client or interface if
     * server(http://www.crossroads.io/1-0:xs-tcp).
     * @param port Port number (as string).
     */
    public UConfig(String source, String port) {
        this.source = source;
        this.port = port;
    }
}
